developing-packages.md

```diff
@@ -0,0 +1,174 @@
+---
+layout: page
+title: Developing Packages & Plugins
+permalink: /developing-packages/
+---
+
+* TOC
+{:toc}
+
+## Package introduction
+
+Dart packages enable the creation of modular code that can easily be shared. A
+minimal package with a single library consists of:
+
+* A `pubspec.yaml` file: A metadata file that declared the package name,
+  version, author, etc.
+
+* A `lib` directory containing the public code in the package, minimally a
+  single `<package-name>.dart` file.
+
+For additional details, see the [Dart library package](https://www.dartlang.org/guides/libraries/create-library-packages) documentation.
+
+### Package types {#types}
+
+Flutter supports several kinds of packages:
+
+* *Dart packages*: General packages written in Dart without any dependencies on
+  any specific frameworks. They can be used across many frameworks such as
+  [AngularDart](https://webdev.dartlang.org/angular) and Flutter. A concrete
+  example is the [`path`](https://pub.dartlang.org/packages/path) package.
+
+* *Flutter packages*: General packages which contain Flutter specific
+  functionality and thus have a dependency on the Flutter framework, restricting
+  their use to Flutter only. A concrete example is the
+  [`fluro`](https://pub.dartlang.org/packages/fluro) package.
+
+* *Flutter plugins*: Flutter packages which in additional to the Dart code also
```

Comment:
I think the package/plugin distinction probably isn't too helpful. A plugin isn't a sub-type of package.

maybe some packages offer plugins, but it's still a package.